### PR TITLE
[KP-737] Fix Rest API

### PR DESCRIPTION
**Pull request by graddaniel**
https://kaholo.atlassian.net/browse/KP-737

### CODE REVIEW

**Comment by graddaniel** [comment]
I forgot to add descriptions to `config.json`

File: app.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  return vaultService.patchSecret(
+    vaultToken,
+    vaultUrl,
+    vaultNamespace,
+    secretsPath,
+    secrets,
+  );
+}
+
+function validateParams(params) {
```
Comment (xkcm) [inline app.js:89]: I think we should put this function in a separate file, **app.js** should contain just functions for plugin's methods.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "lint:fix": "npx eslint . --fix"
+    },
+    "author": "Amit Alon",
+    "license": "ISC",
+    "dependencies": {
+        "eslint": "8.14.0",
+        "eslint-config-airbnb-base": "15.0.0",
+        "eslint-plugin-import": "2.26.0",
+        "kaholo-plugin-library": "github:Kaholo/kaholo-plugin-library#v1.0.3",
+        "node-fetch": "^2.6.7",
```
Comment (xkcm) [inline package.json]: Shouldn't we remove this caret?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "kaholo-plugin-hashicorp-vault",
+    "version": "1.0.6",
```
Comment (xkcm) [inline package.json]: Version from **package.json** should be removed.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    },
+    "author": "Amit Alon",
+    "license": "ISC",
+    "dependencies": {
+        "eslint": "8.14.0",
+        "eslint-config-airbnb-base": "15.0.0",
+        "eslint-plugin-import": "2.26.0",
+        "kaholo-plugin-library": "github:Kaholo/kaholo-plugin-library#v1.0.3",
+        "node-fetch": "^2.6.7",
+        "request": "2.88.2"
```
Comment (xkcm) [inline package.json]: Is this dependency used anywhere?

File: vault.service.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  const response = await fetch(url, options);
+  return response.json();
+}
+
+function handleErrors(endpointData) {
+  const { errors } = endpointData;
+  if (errors) {
+    throw new Error(`Endpoint returned errors: ${errors[0]}`);
```
Comment (xkcm) [inline vault.service.js]: Why not display all errors and not just the first one?

**Review by xkcm — CHANGES_REQUESTED** [review]
Some minor things

File: package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    "scripts": {
+        "lint": "npx eslint .",
+        "lint:fix": "npx eslint . --fix"
+    },
+    "author": "Amit Alon",
+    "license": "ISC",
+    "dependencies": {
+        "eslint": "8.14.0",
+        "eslint-config-airbnb-base": "15.0.0",
+        "eslint-plugin-import": "2.26.0",
```
Comment (xkcm) [inline package.json]: One more thing: **eslint** packages should be in the `devDependencies`, they are not essential for the plugin to work, they are used only in the development.

File: app.js (hunk 1 above)
Comment (graddaniel) [reply app.js:89]: I treat the `app.js` as the layer responsible for data processing, sort of a controller.

Your idea makes sense and could be implemented in future (then we could only import functions from other files, some for autocompletes, some for the methods), when a plugin grows.

File: package.json (hunk 2 above)
Comment (graddaniel) [reply package.json]: Fixed

File: package.json (hunk 3 above)
Comment (graddaniel) [reply package.json]: Fixed

File: package.json (hunk 4 above)
Comment (graddaniel) [reply package.json]: Fixed

File: vault.service.js (hunk 5 above)
Comment (graddaniel) [reply vault.service.js]: Fixed

File: package.json (hunk 6 above)
Comment (graddaniel) [reply package.json]: Fixed

File: app.js (hunk 1 above)
Comment (xkcm) [reply app.js:89]: okay, I see

**Review by xkcm — APPROVED** [review]

**Pull request merged**